### DISCUSSION TITLE
Sharp X68000: How to boot from floppy after a reset?

### DISCUSSION THREAD

**Discussion by HazukiXX** [mamedev/discussions#123] [Q&A]
So, I'm playing Akumajou Dracula and I'd like to be able to soft or hard reset it in the middle of the gameplay. The only way I managed to do so is by entering "UI Controls" mode and pressing F3 or Left Shift+F3. However, doing so sends you to this screen:

![0](https://github.com/user-attachments/assets/b948028b-8ffb-49b1-8187-c7413f197741)

Translation:
"Akumajou Dracula UserDisk Creation / HD Install version 1.00 Copyright 1993 KONAMI"

"This software is only effective when booting from a floppy disk to prevent operational errors.
Reset the computer and boot from "floppy disk 0."

So, I'm guessing mame is booting the system from Hard Disk by default. How can I make it so it will boot from Floppy?

**Comment by angelosa** [comment]
Can't find a manual handy, I speculate a game reset can happen with a button combination if not by physically pressing the NMI reset? (which is unhandled by driver on a quick glance)

**Comment by cracyc** [comment]
The game automatically ejects disk one after it starts so you have to reinsert it when you reset.

**Reply by HazukiXX** [reply]
And what would be the commandline argument for that?

**Reply by sairuk** [reply]
doubt there is one, you probably want to look into leveraging the lua subsystem to detect the reset and reinsert the disk you are after.

https://docs.mamedev.org/luascript/index.html

**Reply by HazukiXX** [reply]
I suppose there's also no way to add an extra command at the initial boot to make it so the system wouldn't automatically eject disk 1 after the game starts?

Alternatively, it seems like the game also has a native soft-reset function. You simply have to press start+select. However, I don't know what are the default mame keys for those.

Pressing only "start" in a real X68000 pauses the game, and pressing Keyboard Number 2 or 3 in mame also does that, so I'm guessing those are the equivalent keys for it in the emulator? (not sure why there are two instead of only one). But I have no idea what the equivalent key for "select" is. Which one is it?

**Reply by cuavas** [reply]
Systems with powered eject floppy drives tend to get very grumpy if disks don’t eject when they should.

**Comment by HazukiXX** [comment]
Well, I guess I'll close this then since there's no apparent solution.
